CMakeLists.txt

```diff
@@ -499,10 +499,19 @@ option(PG_DEBUG_REFS "PG Ref debugging is enabled" OFF)
 
 option(WITH_TESTS "enable the build of ceph-test package scripts/binaries" ON)
 
-option(WITH_FIO "Support for fio engines" OFF)
-if(WITH_FIO)
+# fio
+option(WITH_FIO "build with fio plugin enabled" OFF)
+option(WITH_SYSTEM_FIO "require and build with system fio" OFF)
+if(WITH_SYSTEM_FIO)
   find_package(fio REQUIRED)
-endif(WITH_FIO)
+  if(NOT FIO_FOUND)
```

Comment:
this `if` block is not necessary, as `find_package(fio REQUIRED)` will bail out with an error message if fio is not found. see https://cmake.org/cmake/help/v3.0/command/find_package.html#find-package